libraries/ESP8266WiFi/src/include/ClientContext.h

```diff
@@ -31,6 +31,10 @@ extern "C" void esp_schedule();
 
 #include "DataSource.h"
 
+#define TCP_DEFAULT_KEEPALIVE_IDLE_SEC          7200 // 2 hours
+#define TCP_DEFAULT_KEEPALIVE_INTERVAL_SEC      75   // 75 sec
+#define TCP_DEFAULT_KEEPALIVE_COUNT             9    // fault after 9 failures
```

Comment:
Do you think that releasing 2.4.0 with keepalives enabled by default, after two release candidates didn't have them, may cause some issues? How about adding methods to enable keepalives first, and then enable them by default after the release?

Reply:
I set up an example with 5s idle-keep-alive (default value is 2 hours), logs are:
```
16:41:47.308767 IP 10.43.1.163.23 > 10.43.1.254.51006: Flags [.], ack 28, win 2117, length 0
16:41:47.308797 IP 10.43.1.254.51006 > 10.43.1.163.23: Flags [.], ack 1, win 29200, length 0

16:41:52.808481 IP 10.43.1.163.23 > 10.43.1.254.51006: Flags [.], ack 28, win 2117, length 0
16:41:52.808512 IP 10.43.1.254.51006 > 10.43.1.163.23: Flags [.], ack 1, win 29200, length 0
```
.163 is esp with tcpserial example. We can see that tcp sends empty packets which are answered by peer. ```WiFiClient::connected()``` will answer false once ```keep-alive-count``` packets are not answered. Libraries and Sketchs already take care of ```::connected()``` result, so yes I think there will be no harm. 
I understand and agree that we can never predict side effects. I made this PR so that #2750 can see and test, it can wait 2.5, or I can disable it by default for 2.4.